tor/helpers/threaded_worker.py

```diff
@@ -95,6 +97,13 @@ def parse_json_posts(posts: Dict) -> List:
     return parse_json_posts(result)
 
 
+def is_time_to_scan(config) -> bool:
+    if datetime.now() > config.last_post_scan_time + timedelta(seconds=45):
+        config.last_post_scan_time = datetime.now()
```

Comment:
Having a method named like this would make me think it does not mutate any variable. I see why it was made this way, but I’d suggest separating them out.

Additionally, given how the threads work with python, I’m concerned using config would result in each thread modifying shared memory. Have you looked into registering a global, thread-local variable explicitly?

Reply:
Good point about the method name, I'll fix that.

For threads, this section is not run by the threads. The ThreadPoolExecutor on line 128 is what kicks them off -- when we process this section, we're still inside the primary thread. The individual threads only handle the code referenced from https://github.com/GrafeasGroup/tor/blob/994740b1c68310bdb5e05e1ed6b857697dc64616/tor/helpers/threaded_worker.py#L128-L137, which doesn't include the `config` object.